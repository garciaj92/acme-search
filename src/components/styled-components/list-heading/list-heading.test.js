import React from "react";
import ListHeading from "./list-heading";

import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';

// Mocks
import { consoleErrorFn } from "../../../__mocks__/console.mock";

const props = {
    heading: "Test Heading"
};

describe("ListHeading: Test visual of component", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<ListHeading {...props} />);
    });

    it("renders correctly", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("contains correct href and content", () => {
        expect(wrapper.find("h1").first().text()).toEqual(props.heading);
    });
});

describe("ListHeading: Test prop type validation", () => {
    let consoleErrorSpy;
    beforeEach(() => {
        console.error = consoleErrorFn
        consoleErrorSpy = jest.spyOn(console, "error");
    });

    afterEach(() => {
        consoleErrorSpy.mockClear();
    });

    it("does not throw an error when props are provided", () => {
        shallow(<ListHeading {...props} />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(0);
    });

    it("throws an error when props are not of type `string`", () => {
        shallow(<ListHeading heading={0} />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
    });

    it("throws an error when heading prop is not provided", () => {
        shallow(<ListHeading />);
        expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
    });
});