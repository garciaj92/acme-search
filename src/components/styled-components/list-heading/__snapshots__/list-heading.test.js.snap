// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeading: Test visual of component renders correctly 1`] = `
<h1
  className="font-weight-bold"
>
  Test Heading
</h1>
`;
